components/async-status-mgt/org.wso2.carbon.identity.framework.async.status.mgt/src/main/java/org/wso2/carbon/identity/framework/async/status/mgt/dao/AsyncStatusMgtDAOImpl.java

```diff
@@ -0,0 +1,880 @@
+                    responseContext.setOperationPolicy(
+                            resultSet.getString(SQLConstants.OperationStatusTableColumns.IDN_OPERATION_POLICY));
+                    responseContext.setCreatedTime(
+                            Timestamp.valueOf(
+                                    resultSet.getString(SQLConstants.OperationStatusTableColumns.IDN_CREATED_AT)));
+                    responseContext.setModifiedTime(
+                            Timestamp.valueOf(
+                                    resultSet.getString(SQLConstants.OperationStatusTableColumns.IDN_LAST_MODIFIED)));
+                    responseContexts.add(responseContext);
+                }
+            }
+        } catch (SQLException e) {
+            String errorMessage = "Error fetching async operation status for subject: " + operationSubjectId;
+            LOGGER.info(errorMessage + e);
+            throw new RuntimeException(errorMessage, e);
+        }
+        LOGGER.info("Fetching async operation status for subject: " + operationSubjectId);
+        return responseContexts;
+    }
+
+    @Override
+    public List<ResponseOperationRecord> getOperationRecords(String operationSubjectType, String operationSubjectId,
+                                                             String operationType, Integer limit,
+                                                             List<ExpressionNode> expressionNodes)
+            throws AsyncStatusMgtServerException {
+
+        FilterQueryBuilder filterQueryBuilder = buildFilterQuery(expressionNodes, MODEL_CREATED_TIME);
+        String sqlStmt = getOperationsStatusSqlStmt(filterQueryBuilder);
+
+        List<ResponseOperationRecord> operationRecords;
+        NamedJdbcTemplate namedJdbcTemplate = Utils.getNewTemplate();
+        try {
+            operationRecords = namedJdbcTemplate.executeQuery(sqlStmt,
+                    (resultSet, rowNumber) -> {
+                        ResponseOperationRecord record = new ResponseOperationRecord();
+                        record.setOperationId(resultSet.getString(1));
+                        record.setCorrelationId(resultSet.getString(2));
+                        record.setOperationType(resultSet.getString(3));
+                        record.setOperationSubjectType(resultSet.getString(4));
+                        record.setOperationSubjectId(resultSet.getString(5));
+                        record.setResidentOrgId(resultSet.getString(6));
+                        record.setInitiatorId(resultSet.getString(7));
+                        record.setOperationStatus(resultSet.getString(8));
+                        record.setOperationPolicy(resultSet.getString(9));
+                        record.setCreatedTime(Timestamp.valueOf(resultSet.getString(10)));
+                        record.setModifiedTime(Timestamp.valueOf(resultSet.getString(11)));
+                        return record;
```

Comment:
If we use object builder pattern for this ResponseOperationRecord class, this part can be further improved.